source/material_model/visco_plastic.cc

```diff
@@ -544,6 +666,14 @@ namespace aspect
     void
     ViscoPlastic<dim>::declare_parameters (ParameterHandler &prm)
     {
+      prm.enter_subsection("Compositional fields");
```

Comment:
You do not need to declare a parameter again, this just needs to be done by the plugin owning this parameter.

Reply:
So this, and the next comment are the same problem as above, the `this->n_compositional_fields()` is not available in when it is being tested.